lib/utils/fixEmptyLineBefore.mjs

```diff
@@ -0,0 +1,39 @@
+import { assert } from './validateTypes.mjs';
+
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('postcss').Node} Node */
+/** @typedef {(node: Node, newline: string) => Node} Callback */
+
+/**
+ * fix callback for *-empty-line-before rules
+ * @param {object} o
+ * @param {Node} o.node
+ * @param {string=} o.newline
+ * @param {Callback} o.mutate
+ * @param {Range} [o.range]
+ * @returns {Range}
+ */
+export default function fixEmptyLineBefore({ node, newline, mutate, range }) {
+	let count;
+
+	switch (mutate.name) {
```

Comment:
[suggest] I want to avoid depending on a function name because it's not type-safe. So, I suggest more refactoring as follows:

- Introduce the `action` argument (type: `'add' | 'remove'`) instead of `mutate`.
- Internally call `addEmptyLineBefore()` or `removeEmptyLinesBefore()`.
- If possible, inline `addEmptyLineBefore.mjs` and `removeEmptyLinesBefore.mjs` into this file.

E.g.

```js
switch (action) {
	case 'add':
		count = addEmptyLineBefore(node, newline);
		break;
	case 'remove':
		count = removeEmptyLinesBefore(node, newline);
		break;
	default:
		throw new Error(`Unknown action '${action}'`);
}
```

Reply:
I forgot one point. I'd like to move the logic of `/\r?\n/.test` or `match(/\n/g)` into each add/remove function because of duplication.

Reply:
> If possible, inline addEmptyLineBefore.mjs and removeEmptyLinesBefore.mjs into this file.

I won't do that because these functions have tests.

> I'd like to move the logic of /\r?\n/.test or match(/\n/g) into each add/remove function because of duplication.

This can be done in a different refactor PR.
I would like this PR be about the rule refactor and not the utils refactors.

Reply:
It's also fine to address these refactoring on another PR 👍🏼